### DISCUSSION TITLE
using OPUS

### DISCUSSION THREAD

**Discussion by farzadkabiri-nrp** [ThibaultBee/StreamPack#165] [Q&A]
Hello
Thank you for the wonderful library.
I have written a POC application ( push to talk over cellular) that is a half-duplex .
The only problem I have now is the amount of latency that is not less than 1000 milliseconds. I want to use opus, can you guide me in this matter

**Comment by farzadkabiri-nrp** [comment]
can u help me?

**Reply by ThibaultBee** [reply]
Hi there,

If you are using RTMP, it is supposed to be 2-3 s of latency. In that case you should switch to SRT.
You can use the latency parameter of SRT to reduce it.
Also it worth noticing that StreamPack does not buffer, but your player might be the source of buffering.

I am not familiar with audio only communication but SRT and RTMP are mostly built for videos and there might be some other protocol which are better for audio (RTP?).

**Reply by farzadkabiri-nrp** [reply]
I am using SRT protocol and Exo Player for the player , and currently I have achieved a delay of 800 milliseconds in audio chat and video chat, but it is not enough and I should be able to reduce it to 300 milliseconds.

**Reply by ThibaultBee** [reply]
From what you said, SRT or RTMP protocols do not meet your requirements, you should have a look at WebRTC instead.

**Reply by farzadkabiri-nrp** [reply]
tnx

**Reply by dadangaprasetya** [reply]
Would you be willing to share the pushtotalk project with me? Let's contribute together.
